Assets/GameAssets/World/WorldScripts/Boat/MapShifting.cs

```diff
@@ -0,0 +1,123 @@
+/*****************************************************************************
+// File Name :         MapShifting.cs
+// Author :            Mark Hanson
+// Creation Date :     11/7/2024
+//
+// Brief Description : The array of different boat layouts that are accessible in other scripts
+*****************************************************************************/
+using UnityEngine;
+using UnityEngine.Events;
+using UnityEngine.Serialization;
+
+/// <summary>
+/// The class holding the functionality of map shifting
+/// </summary>
+public class MapShifting : MonoBehaviour
+{
+    //A way of adding layouts into the scene outside of this script
+    [FormerlySerializedAs("_listOfLayouts")]
+    [Tooltip("Add all layouts of boat inside of here")]
+    [SerializeField] private protected GameObject[] _boatLayouts;
+    private protected int _currentLayout = 0;
+    
+    //For calling outside of script into another script
+    private UnityEvent _onMapShifting = new();
+    private UnityEvent _onMapReset = new();
+    
+    /// <summary>
+    /// Allow for first layout to turn on if not on already
+    /// </summary>
+    void Start()
+    {
+        _boatLayouts[_currentLayout].SetActive(true);
+        
+        SubscribeToEvents();
+    }
+    
+    /// <summary>
+    /// Turns off current layout of boat and goes to the next one
+    /// </summary>
+    private protected void LayoutSwapForward()
```

Comment:
None of these functions need to be protected as there is no inheritance going on